### PR TITLE
[pack] Executing performance tests on VM

### LINKED ISSUE #3636: Use VM for performance tests instead of appveyor build
we have 1 hour execution limitation for appveyor build

### PR DESCRIPTION
**Pull request by alrod**
Resolves #3636

### CODE REVIEW

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                _currentExecutionRuntime = runtime;
+            }
+            else
+            {
+                _fixture.Logger.LogInformation($"Execution runtime: {runtime}");
+            }
+        }
+    }
+
+    class TestDefinition
```
Comment (fabiocav) [inline tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs]: Missing access modifier. Consider moving to another file.

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+                var commandResult = await VirtualMachinesOperationsExtensions.RunCommandAsync(_client.VirtualMachines, Settings.SiteResourceGroup, Settings.VM,
+                    new RunCommandInput("RunPowerShellScript",
+                    new List<string>() { $"& 'C:\\Tools\\ps\\test-throughput.ps1' '{test.FileName}' '{test.Desciption}' '{Settings.RuntimeVersion}'" }));
+            }
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
```
Comment (fabiocav) [inline tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs]: nit: remove `this`

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this._disposed)
```
Comment (fabiocav) [inline tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/PerformanceManager.cs]: nit:remove `this`

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.Dashboard/AppVeyorClient.cs [hunk 4]
```diff
@@ -24,53 +24,32 @@ public AppVeyorClient(ILogger logger)
...
         {
-            AppVeyorClient client = new AppVeyorClient(logger);
-            await client.StartBuild(new Build()
-            {
-                Branch = Environment.GetEnvironmentVariable("PerformanceBranch", EnvironmentVariableTarget.Process),
-                ProjectSlug = Environment.GetEnvironmentVariable("PerformanceProjectSlug", EnvironmentVariableTarget.Process),
-                Artifact = "inproc",
-                JobName = "Image: Visual Studio 2017"
-            });
+            this.Dispose(true);
```
Comment (fabiocav) [inline tools/WebJobs.Script.Performance/WebJobs.Script.Performance.Dashboard/AppVeyorClient.cs]: Remove `this`

File: test/WebJobs.Script.Tests.E2E.Shared/FunctionAppFixture.cs [hunk 5]
```diff
@@ -35,17 +35,20 @@ public FunctionAppFixture()
...
                 {
                     Initialize().Wait();
                     return;
                 }
-                catch
+                catch(Exception ex)
                 {
                     // Best effort.
                     System.Threading.Thread.Sleep(5000);
+                    _logger.LogInformation($"Initialize error: {ex.ToString()}");
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.E2E.Shared/FunctionAppFixture.cs]: `.ToString()` is not needed here

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/WebJobs.Script.Performance.App.csproj [hunk 6]
```diff
@@ -1,32 +1,49 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
+    <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp2.1</TargetFramework>
-
-    <IsPackable>false</IsPackable>
+    <IsPackable>true</IsPackable>
+    <PackAsTool>true</PackAsTool>
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply tools/WebJobs.Script.Performance/WebJobs.Script.Performance.App/WebJobs.Script.Performance.App.csproj]: Thanks! I remember the conversation now. It has been a while.

File: tools/WebJobs.Script.Performance/WebJobs.Script.Performance.Dashboard/PerformanceManager.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                var commandResult = VirtualMachinesOperationsExtensions.RunCommand(client.VirtualMachines, siteResourceGroup, vm,
+                    new RunCommandInput("RunPowerShellScript",
+                    new List<string>() { $"& 'C:\\Tools\\ps\\run.ps1' '{appUrl}' '{command}'" }));
+            }
+        }
+    }
+
+    public class PerformanceCommand
+    {
+        public string extensionUrl { get; set; }
```
[the remark this answers is not in the document]
Comment (alrod) [reply tools/WebJobs.Script.Performance/WebJobs.Script.Performance.Dashboard/PerformanceManager.cs]: This class is not needed, I removed it.

**Pull request merged**